<?xml version="1.0"?>

<launch>

    <!-- <arg name="camera" default="camera" /> -->

    <!-- start sensor-->
    <!-- <include file="$(find openni2_launch)/launch/openni2.launch">
        <arg name="camera" default="$(arg camera)"/>
    </include> -->

    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="/zed2i/zed_node/point_cloud/cloud_registered"/>
        <!-- <remap from="scan" to="/base_scan"/> -->
        <rosparam>
            target_frame: zed2i_base_link # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.3
            min_height: -0.25
            max_height: 0.0

            angle_min: -1.5708 # -M_PI/2
            angle_max: 1.5708 # M_PI/2
            angle_increment: 0.0087 # M_PI/360.0
            scan_time: 0.033
            range_min: 0.3
            range_max: 2.0
            use_inf: true
            inf_epsilon: 1.0

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 2
        </rosparam>

    </node>

</launch>
